File: EmployeeManagement/Models/LeaveType.cs
namespace EmployeeManagement.Models;

public class LeaveType
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public int AnnualLimit { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<LeaveRequest> LeaveRequests { get; set; }
        = new List<LeaveRequest>();
}

File: employee-management-client/src/types/leaveType.ts
export interface LeaveType {
    id: string;
    name: string;
    annualLimit: number;
    isActive: boolean;
}

export interface LeaveTypeCreateRequest {
    name: string;
    annualLimit: number;
}

export interface LeaveTypeUpdateRequest {
    name: string;
    annualLimit: number;
}